azurerm/internal/services/compute/shared_image_resource.go

```diff
@@ -112,6 +112,31 @@ func resourceArmSharedImage() *schema.Resource {
 				Optional: true,
 			},
 
+			"purchase_plan": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
```

Comment:
This wants a ForceNew associated with it
```
Code="PropertyChangeNotAllowed" Message="Changing property 'galleryImage.properties.purchasePlan.name' is not allowed." Target="galleryImage.properties.purchasePlan.name"
```

Reply:
Done